test/Microsoft.Azure.WebJobs.Host.UnitTests/ExceptionFormatterTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿using Microsoft.Azure.WebJobs.Host.Diagnostics;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.UnitTests
+{
+    public class ExceptionFormatterTests
+    {
+        [Fact]
+        public void FormatException_RemovesAsyncFrames()
+        {
+            try
+            {
+                var test = new TestClass();
+                test.Run();
+            }
+            catch (Exception exc)
+            {
+                string formattedException = ExceptionFormatter.GetFormattedException(exc);
+
+                Assert.False(Regex.Match(formattedException, "d__.\\.MoveNext()").Success);
```

Comment:
Have some additional tests validating against literal strings for expected traces.